plugin/build.gradle.kts

```diff
@@ -23,6 +23,7 @@ dependencies {
     compileOnly(kotlin("gradle-plugin", PluginVersions.kotlin))
     compileOnly("com.android.tools.build:gradle:${PluginVersions.androidPlugin}")
     implementation("net.swiftzer.semver:semver:${PluginVersions.semver}")
+    implementation("org.eclipse.jgit:org.eclipse.jgit:${PluginVersions.jgit}")
```

Comment:
Does this drag along any transitive dependencies?

Reply:
It brings them:
```
\--- org.eclipse.jgit:org.eclipse.jgit:5.6.0.201912101111-r
     +--- com.jcraft:jsch:0.1.55
     +--- com.jcraft:jzlib:1.1.1
     +--- com.googlecode.javaewah:JavaEWAH:1.1.6
     +--- org.slf4j:slf4j-api:1.7.2
     +--- org.bouncycastle:bcpg-jdk15on:1.64
     |    \--- org.bouncycastle:bcprov-jdk15on:1.64
     +--- org.bouncycastle:bcprov-jdk15on:1.64
     \--- org.bouncycastle:bcpkix-jdk15on:1.64 (*)
```

Reply:
These seems like quite a few dependencies to drag along with the plugin.
If you really think this is necessary, ideally we'd vendor them inside of our own JAR file instead.

Reply:
> If you really think this is necessary, ideally we'd vendor them inside of our own JAR file instead.

Do you mean creating shadowed fat-jar gradle plugin?

Reply:
More or less , that's the idea. Something that re-writes the imports to be something we vendor.

Reply:
I am not a big fan of vendoring in java world as it becomes consumer much harder (ask publisher to do a new release) to update 3rd party dependencies if there are some urgent bugfix

Reply:
I can't find the documentation currently, but it was a common recommendation to try not to bring along dependencies in plugins. Primarily so you significantly reduce the chances that two plugins drag along conflicting dependencies. 

That's quite a few dependencies to be dragging along especially given that we're using such a minor bit of the libraries functionality. If you really, really need to use this dependency, I advise that instead you create a custom configuration so it's loaded on an isolated classpath and call that method via reflection.

Reply:
You convinced me :slightly_smiling_face: 

I will implement fat jar and put there this dependencies as well as semver dependency.

Reply:
That works. Thanks!

Reply:
Done in https://github.com/JLLeitschuh/ktlint-gradle/pull/323/commits/07a26c08362fac95ec5d3037b8c4f80112bab684

It took me a while to combine shadow, java-gradle-plugin and com.gradle.plugin-publish together :disappointed_relieved: 

Happy to hear possible integration improvements.